llvm/lib/ProfileData/InstrProf.cpp

```diff
@@ -428,13 +428,22 @@ std::string getPGOFuncNameVarName(StringRef FuncName,
   return VarName;
 }
 
+bool isGPUProfTarget(const Module &M) {
+  const auto &triple = M.getTargetTriple();
+  return triple.rfind("nvptx", 0) == 0 || triple.rfind("amdgcn", 0) == 0 ||
+         triple.rfind("r600", 0) == 0;
+}
+
```

Comment:
```suggestion
bool isGPUProfTarget(const Module &M) {
  const llvm::Triple &Triple = M.getTargetTriple();
  return Triple.isAMDGPU() || Triple.isNVPTX()
}
```
Standard way looks like this. Side note, we really need a way to express this in a more re-usable way especially with SYCL looming. @arsenm should be make some common interface in `CodeGenModule` that just returns if we're currently targeting a "GPU like" device?

Reply:
I did briefly consider turning the `isGPUProfTarget` function into a method of either the Module or Triple class, but I was worried that would be out of the scope of this PR. Do you think adding an `isGPU` method to the Triple class would make things cleaner?